src/test/scala/com/microsoft/hyperspace/index/rules/RuleUtilsTest.scala

```diff
@@ -112,79 +112,129 @@ class RuleUtilsTest extends HyperspaceRuleTestSuite {
+        }
 
-      // Scenario #1: Append new files.
-      df.write.mode("append").parquet(dataPath)
+        // Scenario #1: Append new files.
+        df.write.mode("append").parquet(dataPath)
 
-      {
-        val optimizedPlan = spark.read.parquet(dataPath).queryExecution.optimizedPlan
-        verify(optimizedPlan, hybridScanEnabled = false, expectCandidateIndex = false)
-        verify(optimizedPlan, hybridScanEnabled = true, expectCandidateIndex = true)
-      }
+        {
+          val optimizedPlan = spark.read.parquet(dataPath).queryExecution.optimizedPlan
+          verify(optimizedPlan, hybridScanEnabled = false, expectCandidateIndex = false)
+          verify(optimizedPlan, hybridScanEnabled = true, expectCandidateIndex = true)
+        }
 
-      // Scenario #2: Delete 1 file.
-      {
-        val readDf = spark.read.parquet(dataPath)
-        readDf.queryExecution.optimizedPlan foreach {
-          case LogicalRelation(
-              HadoopFsRelation(location: PartitioningAwareFileIndex, _, _, _, _, _),
-              _,
-              _,
-              _) =>
-            systemPath
-              .getFileSystem(new Configuration)
-              .delete(location.allFiles.head.getPath, false)
-          case _ =>
+        // Scenario #2: Delete 1 file.
+        {
+          FileUtils.delete(new Path(indexFile), isRecursive = false)
         }
-      }
 
-      {
-        val optimizedPlan = spark.read.parquet(dataPath).queryExecution.optimizedPlan
-        verify(optimizedPlan, hybridScanEnabled = false, expectCandidateIndex = false)
-        verify(optimizedPlan, hybridScanEnabled = true, expectCandidateIndex = true)
-      }
+        {
+          val optimizedPlan = spark.read.parquet(dataPath).queryExecution.optimizedPlan
+          verify(optimizedPlan, hybridScanEnabled = false, expectCandidateIndex = false)
+          // TODO: expectedCandidateIndex = true once delete dataset is supported.
+          verify(optimizedPlan, hybridScanEnabled = true, expectCandidateIndex = false)
+        }
 
-      // Scenario #3: Replace all files.
-      df.write.mode("overwrite").parquet(dataPath)
+        // Scenario #3: Replace all files.
+        df.write.mode("overwrite").parquet(dataPath)
 
-      {
-        val optimizedPlan = spark.read.parquet(dataPath).queryExecution.optimizedPlan
-        verify(optimizedPlan, hybridScanEnabled = false, expectCandidateIndex = false)
-        verify(optimizedPlan, hybridScanEnabled = true, expectCandidateIndex = false)
+        {
+          val optimizedPlan = spark.read.parquet(dataPath).queryExecution.optimizedPlan
+          verify(optimizedPlan, hybridScanEnabled = false, expectCandidateIndex = false)
+          verify(optimizedPlan, hybridScanEnabled = true, expectCandidateIndex = false)
+        }
       }
     }
   }
 
+  test("Verify the location of injected shuffle for Hybrid Scan.") {
+    withTempDir { tempDir =>
+      val dataPath = tempDir + "/hbtable"
```

Comment:
you can just do `withTempDir { dataPath =>` right? (No need for "/hbtable")

Reply:
Yes but it still requires `toString`

Reply:
Ah, it fails since the directory is already created. You need `withTempPath` (which is same implemenation as `withTempDir` except that it deletes the directory created). I pushed the changes.

Btw, `parquet()` works on a directory (if you do overwrite).